### PR TITLE
Add declarative auth support for gRPC client

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$title

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1465

### CODE REVIEW

**Review by ldclakmal — DISMISSED** [review]
Reviewed the implementation logic and it's LGTM.

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

File: ballerina/client_endpoint.bal [hunk 1]
```diff
@@ -154,6 +176,38 @@ isolated function generateMethodId(string? pkgName, string svcName, string rpcNa
...
+    } else {
+        // Here, `authConfig` is `OAuth2GrantConfig`
+        ClientOAuth2Handler handler = new(authConfig);
+        return handler;
+    }
+}
+
+// Enriches the request using the relevant client auth handler
+isolated function enrichHeaders(ClientAuthHandler clientAuthHandler, map<string|string[]> headers) returns map<string|string[]>|ClientAuthError {
+    if (clientAuthHandler is ClientBasicAuthHandler) {
```
Comment (BuddhiWathsala) [inline ballerina/client_endpoint.bal:199]: Shall we remove these unnecessary parentheses?

Comment (dilanSachi) [reply ballerina/client_endpoint.bal]: Removed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/350/commits/bba296b92881ca062ad2b802684cc6f690bcba4f

**Review by BuddhiWathsala — APPROVED** [review]

**Pull request merged**